### PR TITLE
GCI: Made student section dynamic and resolved the bug, added 'Abishek V Ashok' to student section.

### PR DESCRIPTION
**Pull request by abishekvashok**
Some student messed up with the code of the index.html, I now fixed it
and replaced it with jekyll code, now jekyll will import one by one by
one block of student from student.yml file just like it is done for
mentors secton and blog section.

then as part of my task here:
https://codein.withgoogle.com/dashboard/task-instances/5169781940420608/

I added my info to it

Thank you, happy coding!

### CODE REVIEW

**Comment by abishekvashok** [comment]
Ok @jig08 take a look madam. It looks fine but has conflicts let me see how to resolve them.

**Comment by abishekvashok** [comment]
@jig08 after running the commands in the checkout via cmd line, I am getting Everything is up to date but nothing is working out help please... Or, should I make a new PR closing this one and forking the repo again!

**Comment by Princu7** [comment]
@Abhi2424shek Did you pull new changes from the main repository yet? If not, then please run
```
git remote add origin https://github.com/fossasia/gci16.fossasia.org.git
git pull origin gh-pages
```
Then try making a pull request.
Also watch this video for removing the merge conflicts
https://www.youtube.com/watch?v=g8BRcB9NLp4

**Comment by abishekvashok** [comment]
@Princu7 should i close the PR and start a new one, running the code gives: Invalid refspec: gh-pages

**Comment by Princu7** [comment]
@Abhi2424shek Maybe try directly running the command `git pull`. Closing PR and reforking the repo again might do the job but it is not a good practice to follow.

**Comment by abishekvashok** [comment]
@Princu7 let me try out

**Comment by abishekvashok** [comment]
I have squashed all the commits into a single commit so please merge it, the squashed commit is named "Solved as stated" so please help out and approve my task. I've done pretty more with this PR...
@jig08 @Princu7 .

**Comment by abishekvashok** [comment]
@jig08 @Princu7 now I have done everything I could, now it's up to you to squash all the commits and merge (via merge pull request or simply merge the pull request). Please do so before conflicts arise as I have modified huge amount of core files. I've also submitted the task for review. Thank you...

**Comment by Princu7** [comment]
@Abhi2424shek Please post a link to your gh-pages for easy reviewing. Thanks!!

**Comment by abishekvashok** [comment]
@Princu7 there are some conflicts let me resolve them, squash and I will send the current and my development gh-pages screen shots for easy reviewing

**Comment by abishekvashok** [comment]
@Princu7 @jig08 @mariobehling , now is the time go ahead and merge it...
Here is the preview: [](https://abhi2424shek.github.io/gci16.fossasia.org/)
See the students section, please.

Thanks, happy coding

**Comment by abishekvashok** [comment]
Help out quickly guys, before more conflicts arise as i have modified the core files...

**Comment by abishekvashok** [comment]
@hemantjadon can you help out sir...

**Comment by hemantjadon** [comment]
@Abhi2424shek 
Hi you will have to squash the commits before we can proceed forward and merge.
For resources on squashing the commits please check the slack channel "#gci" and see the pinned messages.

**Comment by abishekvashok** [comment]
@hemantjadon can you send me the link

**Comment by Princu7** [comment]
@Abhi2424shek Sorry, I was busy. Here is the link
https://fossasia.slack.com/messages/gci/
Sorry, but I can't see your gh-pages link. Can you please post it again? Thanks!!

**Comment by abishekvashok** [comment]
Hey @hemantjadon @Princu7 after running git rebase -i I am getting commits that I have'nt made !

**Comment by abishekvashok** [comment]
I squashed @hemantjadon @Princu7. Now merge while adding the merge commit, please

**Review by daminisatya — CHANGES_REQUESTED** [review]
No you dint squash yet!

**Comment by abishekvashok** [comment]
I cant squash  more as it will leave conflicts again and again @daminisatya . Further the two commits are empty

**Comment by daminisatya** [comment]
@Abhi2424shek You need to squash the top two commits. Follow the steps
```git rebase -i HEAD~2``` Which means you are taking the first two from the stack. 

Now a commad promt shows the two commit messages, select the second one as squash and first one as pick. Read the instructions on that screen clearly, then another screen pops out giving you an option to make a commit description. and then you need to force push.

```git push -f origin gh-pages``` or what ever branch you are working on.

**Comment by abishekvashok** [comment]
@daminisatya this is what I see:
`pick cce357d Fixed studetn import bug, added abishek v ashok to student section.
pick 1ede1a9 not sure how to link, so I'm uploading a 2nd version
pick 5e5edc8 Added 2 mentors
pick a0ce48c Updated travis.yml
pick bf32256 Travis-CI: Allow building any branch name
pick 915ed68 Fix linting errors
pick c620aa0 Update students.yml
pick 3a3e8d5 Improved contributors section
pick 2cf19db Fixed inconsistencies
pick 4634aed Add student creds to html students section`

**Comment by daminisatya** [comment]
Give a screenshot. It's convoluted.

**Comment by daminisatya** [comment]
Also share me the screenshot after typing this

```git log```

**Comment by abishekvashok** [comment]
@daminisatya I am seeing more than 2 commits!

**Comment by abishekvashok** [comment]
image

@daminisatya , the git log

**Comment by abishekvashok** [comment]
@daminisatya when I run
`
`git rebase -i

and in the notepad type in:
pick myfirstcommit has
squash mysecond commit has`
`
I am getting some cherry empty errors!

**Comment by daminisatya** [comment]
```git rebase -i HEAD~2``` This is the command for your log

**Comment by abishekvashok** [comment]
cant you squash them in the github merge pull request @daminisatya

**Comment by abishekvashok** [comment]
@daminisatya heres what I get, @Princu7 @hemantjadon take a look please, @mariobehling
image

**Comment by abishekvashok** [comment]
@hemantjadon could you take a look, why dont you squash and merge (in the merge button)

**Comment by hemantjadon** [comment]
@Abhi2424shek 

* The merge button merges the PR, and does't give us opportunity to squash.

* You can simply squash the top two commits into then we can proceed merging, as @daminisatya has mentioned.

**Comment by abishekvashok** [comment]
@hemantjadon see the down button near, it has the option i guess and rebasing causes me conflicts, shall I start a new PR

**Comment by hemantjadon** [comment]
@daminisatya It looks good from my side (despite of two blank commits) which can be solved using squashing while merging. Your inputs and review please ?

**Comment by abishekvashok** [comment]
@hemantjadon should i stop trying to squash or continue. I am running:

`pick cce357d Fixed studetn import bug, added abishek v ashok to student section.`
`squash b0f2a4e Resolved commits`
`squash 7171a12 Merge branch 'Abhi2424shek-gh-pag' into gh-pages`

I cant squash running above (after git rebase -i -p HEAD~2) gives cherry-pick failed error...

**Comment by abishekvashok** [comment]
@hemantjadon @Princu7 @jig08 @daminisatya , can someone now squash and merge out, so that I could work on the next task in GCI... and unleash my contributions :)

**Comment by abishekvashok** [comment]
Some one help out...

**Comment by abishekvashok** [comment]
@hemantjadon i could wait

**Comment by Princu7** [comment]
@Abhi2424shek Thanks for your patience and understanding. I know it's exhausting and frustrating to wait but there are many things that have to be kept in mind before merging.

Also, If you have a problem that you can't work on other tasks without getting the current task approved since you have made changes in the master branch, then you can alter your workflow a little. From next time, you can make a separate branch for every new task you are going to do. Then you can push that branch to your forked repository and generate pull requests from there. That way, you can work on more than one task at the same time and delete the branches after the tasks are approved.
Here is the link which describes doing so in a simple manner. Do give it a read. It may help you. Thanks
https://gun.io/blog/how-to-github-fork-branch-and-pull-request/

**Comment by abishekvashok** [comment]
@Princu7 can you merge mine now! see the preview: 
image
image

and this is the current one at (gci16.fossasia.org)
image

**Comment by Princu7** [comment]
@Abhi2424shek Hehe. Don't worry. Your PR will be merged as soon as @daminisatya reviews and approves your changes. Cheers!!

**Comment by abishekvashok** [comment]
@Princu7 i wait for then @daminisatya , k cheeres :) !!! But i worry if she is late I would have conflicts.... This issue has to be resolved fast as plenty of students are going the static way to integrate themselves to the students section...

**Comment by abishekvashok** [comment]
@hemantjadon @jig08 can you now review

**Comment by abishekvashok** [comment]
@jig08 can you help out @hemantjadon @Princu7

**Comment by abishekvashok** [comment]
@jig08 let resolve the conflicts

**Comment by abishekvashok** [comment]
@jig08 @Princu7 @mariobehling @hemantjadon @daminisatya Please squash and merge this PR before more problems arise :) , it's been a while and no responses :) 💯

File: index.html
Comment (hemantjadon) [inline index.html:237]: Remove these Merge conflict tokens.

**Review by hemantjadon — CHANGES_REQUESTED** [review]

**Comment by abishekvashok** [comment]
This is really complex now I will start a new PR

**Pull request closed without merging**
